### PR TITLE
Add changelog doc & update PR template

### PR DESCRIPTION
**Pull request by akinsolb**
## Change list

Please provide briefly described change list which are you going to propose. 
 
## Types of changes

What types of changes are you proposing/introducing to .NET client?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Integration tests
- [ ] Have you provided integration tests to pass against the latest version of appium? (for Bugfix or New feature)

## Details

Moved release notes to a changelog document so it doesn't reflect in nuget upon each release/
Updated pull request template to encourage documentation  of new features as part of the process

Please provide more details about changes if it is necessary. If there are new features you can provide code samples which show the way they
work and possible use cases. Also you can create [gists](https://gist.github.com) with pasted C# code samples or put them here using markdown. 
About markdown please read [Mastering markdown](https://guides.github.com/features/mastering-markdown/) and [Writing on GitHub](https://help.github.com/categories/writing-on-github/)

### CODE REVIEW

**Comment by akinsolb** [comment]
Just need **anybody** to approve this before I make the release

File: CHANGELOG.MD [hunk 1]
```diff
@@ -0,0 +1,214 @@
+## 4.0
+- Added the ability to terminate a given application on the device (#340). See [documentation](http://appium.io/docs/en/commands/device/app/terminate-app)
+- Added the ability to query the given app status on a device (#341). See [documentation](http://appium.io/docs/en/commands/device/app/app-state/)
```
Comment (mykola-mokhnach) [inline CHANGELOG.MD]: a device or the device?

File: CHANGELOG.MD [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+- [Enhancement] Update support for backgrounding (android) app (#292)
+- [Enhancement] Add file push mechanism for ios and android driver (#301)
+- [Enhancement] Add windows support for page Object pattern (#276)
+- [Enhancement] Redirect local appium server log to Visual Studio output (#303)
+- [BUG FIX] Override IFindsBy methods in appium driver (#302) fixes #265
+- [BUG FIX] Refactor - removing the use of reflection (#297) fixes #291
+- [BUG FIX] Allow log file to be parsed  appium log file  (#286)
+
+### 4.0.0.4-beta
+- [BUG FIX] Fix for w3C locator strategies
```
Comment (mykola-mokhnach) [inline CHANGELOG.MD]: W3C

File: CHANGELOG.MD (hunk 1 above)
Comment (akinsolb) [reply CHANGELOG.MD]: Copy and paste from appium.io. Just a sec

**Pull request merged**